### PR TITLE
Support grow by more than 1 in resizeAllocRange

### PR DESCRIPTION
**Pull request by ben-albrecht**
`resizeAllocRange` only expected a difference of `1` between range sizes `r` and `r2`.  Consequently, it assumes `r.size+1` or `r.size*1.5` will be sufficiently large to accommodate `r2`.

This worked previously, because all the calls to `resizeAllocRange` only had a size delta of `1`. However, In #7180 we need to make bulk array resizes, which breaks down this assumption.

This PR modifies `resizeAllocRange` to account for `r2.size` when resizing, to accommodate any range delta `> 1`.

I also added a test for this functionality.

## Testing

- [x] linux64

### CODE REVIEW

**Comment by ben-albrecht** [comment]
@daviditen - requesting review.

Something that occurred to me when making this change and adding tests:

Should `resizeAllocRange` inherit the value of `r` from `this._value.dataAllocRange`? If not, should it be a private helper function, since it doesn't access/modify any array fields?

File: test/arrays/methods/internal/resizeAllocRange.chpl [hunk 1]
```diff
@@ -0,0 +1,33 @@
+/* Test to exercise the Grow code paths in _array.resizeAllocRange */
```
Comment (daviditen) [inline test/arrays/methods/internal/resizeAllocRange.chpl]: As a hedge against arrayAsVecGrowthFactor changing in the future and breaking this test you could explicitly set it to 1.5 in a .compopts file.

Comment (ben-albrecht) [reply test/arrays/methods/internal/resizeAllocRange.chpl]: good call

Comment (daviditen) [reply test/arrays/methods/internal/resizeAllocRange.chpl]: This looks good to me assuming your testing is clean.

Comment (ben-albrecht) [reply test/arrays/methods/internal/resizeAllocRange.chpl]: Cool - paratesting now!

**Comment by ben-albrecht** [comment]
> Should resizeAllocRange inherit the value of r from this._value.dataAllocRange? 

Offline, @daviditen responded: 👍 

I will make this change in #7180 along with some other refactoring.

**Pull request merged**